content/blog/accessing-hpe-data-fabric-s3-storage-from-spring-boot-s3-micro-service-deployed-in-k3s-cluster.md

```diff
@@ -0,0 +1,155 @@
+
+3.      Java 11 , Apache Maven 3.8+ , Docker Client.
+
+**Build and Install Steps**
+
+1.      Check out  an existing Spring Boot application from [github](<https://github.hpe.com/kiran-mavatoor/df-s3-springboot-k3s-demo>).
+
+2.      Copy the ssl_usertruststore.p12 from the HPE Data Fabric cluster into certs folder under project directory. The ssl_usertruststore.p12 file is located at /opt/mapr/conf directory in cluster node. The password for p12 can be copied from “ssl.client.truststore.password” property value in /opt/mapr/conf/ssl-client.xml .
+
+3.      From project directory, open resources/application.properties. Change the key values as per your environment. 
+
+4.      Execute “mvn clean install” .
+
+5.      The distributable is available in target/df-s3-springboot-k3s-demo-1.0-SNAPSHOT.jar .
+
+6.      Edit the DockerFile located in project directory. The value of 
+
+“-Djavax.net.ssl.trustStorePassword”  must be same as the value of  “ssl.client.truststore.password” obtained from step 2. 
+
+Note: This value can be configured using config-map.yaml in K3s cluster.
+
+7.      Execute below docker commands to build docker image and push it to docker hub. 
+
+Note: Alternatively , we can use podman instead of docker to create images. More information on podman can be obtained from [here](https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli) [](https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli)
+
+```bash
+docker build -f Dockerfile -t <Dockerhub user id>/df-s3-springboot-k3s-demo:latest .
+docker image ls
+docker login -u <Dockerhub user id>
+> enter password:  <Dockerhub password>
+docker push <Dockerhub user id>/df-s3-springboot-k3s-demo:latest
+
+```
+
+8.      Create below df-s3-springboot-k3s-demo.yaml file for deploying the executable in K3s cluster. Sample yaml file is given in project directory. Please replace <dockerhub userid> with valid id.
+
+```Yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: df-s3-springboot-k3s-demo-service
+spec:
+  selector:
+    app: df-s3-springboot-k3s-demo
+  ports:
+  - protocol: TCP
+    name: df-s3-springboot-k3s-demo
+    port: 8000
+    targetPort: 8000
+  type: LoadBalancer
+
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: df-s3-springboot-k3s-demo
+spec:
+  selector:
+    matchLabels:
+      app: df-s3-springboot-k3s-demo
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: df-s3-springboot-k3s-demo
+    spec:
+      containers:
+      - name: df-s3-springboot-k3s-demo
+        image: <Dockerhub userid>/df-s3-springboot-k3s-demo:latest
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 8000
+```
+
+9.      Before deploying it in Kubernetes cluster, validate the docker or podman image by running the image. 
```

Comment:
Add the article "the" prior to "Kubernetes cluster,"

Reply:
done